.github/workflows/release.yml

```diff
@@ -39,19 +29,54 @@ jobs:
           secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials registry | PRIME_REGISTRY ;
           secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials username | PRIME_REGISTRY_USERNAME ;
           secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials password | PRIME_REGISTRY_PASSWORD
-    - name: Build and push all image variations
+    - name: Check SemVer Characteristics
+      id: semver_check
+      run: |
+        tag="${{ github.ref_name }}"
+        if [[ "$tag" == v* ]]; then
+          tag="${tag:1}"
+        fi
+        has_prerelease() { [[ "$1" =~ ^[0-9]+\.[0-9]+\.[0-9]+- ]]; }
+        has_build_meta() { [[ "$1" =~ \+ ]]; }
+
+        if has_prerelease "$tag"; then
+          echo "HAS_PRERELEASE=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "HAS_PRERELEASE=false" >> "$GITHUB_OUTPUT"
+        fi
+
+        if has_build_meta "$tag"; then
+          echo "HAS_BUILD_META=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "HAS_BUILD_META=false" >> "$GITHUB_OUTPUT"
+        fi
+    - name: Build and push image (DockerHub and PrimeStaging)
       uses: rancher/ecm-distro-tools/actions/publish-image@master
       with:
         image: shell
-        tag: ${{ github.ref_name }}${{ matrix.tag-suffix }}
-        platforms: ${{ matrix.platforms }}
+        tag: ${{ github.ref_name }}
+        platforms: "linux/amd64,linux/arm64"
 
         public-registry: docker.io
         public-repo: rancher
         public-username: ${{ env.DOCKER_USERNAME }}
         public-password: ${{ env.DOCKER_PASSWORD }}
 
-        prime-registry: ${{ env.PRIME_REGISTRY }}
+        prime-registry: ${{ env.PRIME_STAGING_REGISTRY }}
         prime-repo: rancher
         prime-username: ${{ env.PRIME_REGISTRY_USERNAME }}
```

Comment:
This is the merge blocker - as we may not have access to push to Staging with the current credentials. So we'll hold on merge until we get confirmation on scope of the Registry credentials.